### PR TITLE
#69 Gates

### PR DESCRIPTION
**Pull request by MrJackey**
## Summary
Fixes # 69
Renamed doors to gates.
Gates can be opened by signal emitters.

### CODE REVIEW

File: Assets/Scripts/Puzzle/ExitGate.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+
+public class ExitGate : Gate {
+	[Space]
```
Comment (smeas) [inline Assets/Scripts/Puzzle/ExitGate.cs]: Space?

File: Assets/Scripts/Puzzle/ExitGate.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+
+public class ExitGate : Gate {
```
Comment (smeas) [inline Assets/Scripts/Puzzle/ExitGate.cs]: The only thing that makes this an **exit** gate is the name of the event: `onLevelComplete`. Maybe rename it to something more generic, or move this behaviour into the `Gate` class?

File: Assets/Scripts/Puzzle/ExitGate.cs (hunk 1 above)
Comment (MrJackey) [reply Assets/Scripts/Puzzle/ExitGate.cs]: When writing this I was expecting a bit more to this script and wanted to separate it from the normal gate. Could probably be removed.

File: Assets/Scripts/Puzzle/ExitGate.cs (hunk 2 above)
Comment (MrJackey) [reply Assets/Scripts/Puzzle/ExitGate.cs]: Ye, true. Still think it should be it's own gate though. Any good thougths on names?

Comment (smeas) [reply Assets/Scripts/Puzzle/ExitGate.cs]: No, not really: GenericGate, TheGate, UsableGate, ActivatableGate, GateGate, DoorGate, Gaterade, Gejt, SIgnalGate, EventGate, HeavensGate.

**Review by smeas — APPROVED** [review]
Gejt!

**Review by Addehman — APPROVED** [review]
I approve!

**Pull request merged**